flang/lib/Optimizer/Transforms/AffineDemotion.cpp

```diff
@@ -77,6 +77,14 @@ class ConvertConversion : public mlir::OpRewritePattern<fir::ConvertOp> {
   matchAndRewrite(fir::ConvertOp op,
                   mlir::PatternRewriter &rewriter) const override {
     if (op.res().getType().isa<mlir::MemRefType>()) {
+      if (auto refTy = op.value().getType().dyn_cast<fir::ReferenceType>())
+        if (auto arrTy = refTy.getEleTy().dyn_cast<fir::SequenceType>()) {
+          fir::SequenceType::Shape flatShape = {-1};
```

Comment:
Should use the function to get the unknown value instead of a constant -1.

Reply:
done, thank you.